data/page-scraper-content-script.js

```diff
@@ -0,0 +1,14 @@
+function sendData() {
+  const text = document.documentElement.innerHTML;
+  const url = document.documentURI;
+  self.port.emit("message", {type: "PAGE_HTML", data: {text, url}});
+}
+
+sendData();
+if (document.readyState === "complete") {
+  sendData();
```

Comment:
Food for thought: you might want to wait a bit before sending data. That's to account for dynamically generated pages. Something like ~3 secs should be reasonable.